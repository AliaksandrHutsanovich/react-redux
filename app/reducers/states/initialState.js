export const initialState = {
  categories: [
    {
      title: 'Mathematic',
      subCategories: [
        {
          title: 'highMathematic',
          subCategories: [],
          tasks: [
            {
              title: 'integral',
              Description: 'find',
              isFinished: false,
            },
            {
              title: 'different',
              Description: 'find',
              isFinished: false,
            },
          ],
        },
        {
          title: 'lowMathematic',
          subCategories: [],
          tasks: [
            {
              title: 'table',
              Description: 'constit',
              isFinished: true,
            },
            {
              title: 'multi',
              Description: 'count',
              isFinished: false,
            },
          ],
        },
      ],
      tasks: [
        {
          title: 'graph',
          Description: 'draw',
          isFinished: true,
        },
        {
          title: 'counting',
          Description: 'count',
          isFinished: true,
        },
      ],
    },
    {
      title: 'Geography',
      subCategories: [
        {
          title: 'highGeography',
          subCategories: [
            {
              title: 'smallHighGeography',
              subCategories: [],
              tasks: [
                {
                  title: 'peoples',
                  Description: 'study',
                  isFinished: true,
                },
                {
                  title: 'capitals',
                  Description: 'study',
                  isFinished: false,
                },
              ],
            },
          ],
          tasks: [
            {
              title: 'polit',
              Description: 'study',
              isFinished: false,
            },
            {
              title: 'mountain',
              Description: 'study',
              isFinished: false,
            },
          ],
        },
      ],
      tasks: [
        {
          title: 'map',
          Description: 'study',
          isFinished: false,
        },
        {
          title: 'drawing',
          Description: 'draw',
          isFinished: false,
        },
      ],
    },
  ],
};

export default initialState;
