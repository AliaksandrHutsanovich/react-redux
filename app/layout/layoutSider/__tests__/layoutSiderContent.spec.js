import React from 'react';
import { mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import { Map } from 'immutable';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import { initialState as store } from '../../../reducers/states/initialState';
import LayoutSiderContent from '../layoutSiderContent';

const mockStore = configureStore();
const initialState = Map().merge(store);

describe('Unit test of layout sider content component', () => {
  const dataStore = mockStore({ actionReducers: initialState });
  const Component = mount(
    <Provider store={dataStore}>
      <BrowserRouter>
        <LayoutSiderContent />
      </BrowserRouter>
    </Provider>,
  );
  it('Full render test', () => {
    expect(Component).toMatchSnapshot();
  });
});
