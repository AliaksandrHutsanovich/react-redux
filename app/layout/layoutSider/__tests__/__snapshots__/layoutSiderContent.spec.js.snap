// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit test of layout sider content component Full render test 1`] = `
<div>
  <Connect(Component)
    placeholder="Add new category"
  />
  <Connect(Component) />
</div>
`;
